topic-operator/src/main/java/io/strimzi/operator/topic/v2/ReplicasChangeClient.java

```diff
@@ -0,0 +1,364 @@
+
+/**
+ * Replicas change client for Cruise Control API.
+ * <br/><br/>
+ * The REST endpoints are {@code topic_configuration} to request replication factor changes and {@code user_tasks} 
+ * to check the asynchronous execution result. Cruise Control runs one task execution at a time, additional requests 
+ * are queued up to {@code max.active.user.tasks}.
+ * <br/><br/>
+ * At any given time, a KafkaTopic.spec.replicas change can be in one of the following states:
+ * <br/>
+ * <ul><li>Pending: Not in Cruise Control's task queue (not yet sent or request error).</li>
+ * <li>Ongoing: In Cruise Control's task queue, but execution not started, or not completed.</li>
+ * <li>Completed: Cruise Control's task execution completed (target replication factor reconciled).</li></ul>
+ */
+public class ReplicasChangeClient {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(ReplicasChangeClient.class);
+    private static final String USER_TASK_ID_HEADER = "User-Task-ID";
+    private static final long REQUEST_TIMEOUT_SEC = 60;
+    
+    private final TopicOperatorConfig config;
+    private final ExecutorService httpClientExecutor;
+    private final ObjectMapper mapper;
+    
+    /**
+     * Create a new replicas change client instance.
+     * 
+     * @param config Topic Operator configuration.
+     */
+    public ReplicasChangeClient(TopicOperatorConfig config) {
+        this.config = config;
+        this.httpClientExecutor = Executors.newCachedThreadPool();
+        this.mapper = new ObjectMapper();
+    }
+
+    /**
+     * Send a topic_configuration request to create a task for replication factor change of one or more topics.
+     * This should be called when one ore more .spec.replicas changes are detected.
+     * 
+     * @param reconcilableTopics Pending replicas changes.
+     * @return Replicas changes with status update.
+     */
+    public List<ReconcilableTopic> requestPendingChanges(List<ReconcilableTopic> reconcilableTopics) {
+        List<ReconcilableTopic> result = new ArrayList<>();
+        if (reconcilableTopics.isEmpty()) return result;
+        updateToPending(reconcilableTopics, "Replicas change pending");
+        result.addAll(reconcilableTopics);
+        
+        try {
+            LOGGER.debugOp("Sending topic configuration request, topics {}", topicNames(reconcilableTopics));
+            HttpClient client = buildHttpClient();
+            Map<Integer, List<ReconcilableTopic>> topicsByReplicas = reconcilableTopics.stream().collect(groupingBy(rt -> rt.kt().getSpec().getReplicas()));
+            client.sendAsync(buildTopicConfigPostRequest(topicsByReplicas), HttpResponse.BodyHandlers.ofString())
+                .thenAccept(response -> {
+                    if (response.statusCode() != 200 || response.headers() == null) {
+                        updateToFailed(reconcilableTopics, format("Replicas change failed (%s)", response.statusCode()), response);
```

Comment:
Given the other logic, there is no need to check for empty headers.

Reply:
But you're checking ` || response.headers() == null` in the line above to avoid an NPE in the other logic. In that case the message would be "Replicas change failed (200)", which looks like a success. Sure you can treat null headers the same as empty, but that's not what you're doing right now.

Reply:
I mean we can remove the null check, as it seems unlikely to receive no header at all. Wdyt?